src/Illuminate/Database/Connectors/PostgresConnector.php

```diff
@@ -81,33 +81,41 @@ protected function configureTimezone($connection, array $config)
     }
 
     /**
-     * Set the schema on the connection.
+     * Set the search_path on the connection.
      *
      * @param  \PDO  $connection
      * @param  array  $config
      * @return void
      */
-    protected function configureSchema($connection, $config)
+    protected function configureSearchPath($connection, $config)
     {
-        if (isset($config['schema'])) {
-            $schema = $this->formatSchema($config['schema']);
+        if (isset($config['searchpath']) || isset($config['schema'])) {
+            $searchPath = $this->formatSearchPath($config['searchpath'] ?? $config['schema'] ?? '"public"');
 
-            $connection->prepare("set search_path to {$schema}")->execute();
+            $connection->prepare("set search_path to {$searchPath}")->execute();
         }
     }
 
     /**
-     * Format the schema for the DSN.
+     * Format the search path for the DSN.
      *
-     * @param  array|string  $schema
+     * @param  array|string  $searchPath
      * @return string
      */
-    protected function formatSchema($schema)
+    protected function formatSearchPath($searchPath)
     {
-        if (is_array($schema)) {
-            return '"'.implode('", "', $schema).'"';
+        if (is_array($searchPath)) {
+            $schemas = array_map(function($item) {
+                return '"'.$item.'"';
+            }, $searchPath);
+
+            $preparedSchemas = array_map(function($item) {
+                return ":$item";
+            }, $schemas);
+
+            return implode(', ', array_combine($preparedSchemas, $schemas));
         } else {
```

Comment:
this else statement is unnecessary

Reply:
Not if we want the existing tests to pass...

Reply:
I believe what he means is that you could keep the code inside of the else block but remove the else keyword itself and the functionality would remain the same. eg:

```
if (is_array($searchPath)) {
    ...
    return implode(', ', array_combine($preparedSchemas, $schemas));
}

return implode(', ', [":$searchPath" => '"'.$searchPath.'"']);
```

@cbj4074

Reply:
@dlett Aha! Of course that's what he meant. Been a long few days... thanks!

Reply:
This is exactly the reason to provide more clear suggestions and examples. Thanks for clearing that up @dlett . I do agree that this style makes more since.

However, this is the style preference from the original ....

```php
/**
     * Format the schema for the DSN.
     *
     * @param  array|string  $schema
     * @return string
     */
    protected function formatSchema($schema)
    {
        if (is_array($schema)) {
            return '"'.implode('", "', $schema).'"';
        } else {
            return '"'.$schema.'"';
        }
    }
```

Reference: https://github.com/laravel/framework/blob/5.5/src/Illuminate/Database/Connectors/PostgresConnector.php#L109

Not sure why this isn't merged yet to be honest. 

Any serious discussion in regards to why this should not be merged ??

Reply:
Things take a little time to get merged sometimes. There is no specific reason. It's just nature of open source :smile:

Reply:
@taylorotwell sure, I understand that completely.

My concern was the hair splitting discussions around personal style preferences with no relevant discussions on actual concerns with the ability to merge this proposal. I have no issues making changes to satisfy any concerns, but they should be justified.

I don't mean to offend anyone involved. Just my 2 cents ;)

Reply:
And this is precisely why I left it alone: https://github.com/laravel/framework/pull/22425

Reply:
Hopefully, everybody is happy now. You win, @jmarcher and @carusogabriel . :clap: